p2p/src/net/default_backend/peer.rs

```diff
@@ -114,22 +114,39 @@ where
         }
     }
 
-    async fn handshake(&mut self) -> crate::Result<()> {
+    fn validate_peer_time(
+        p2p_config: &P2pConfig,
+        local_time: BlockTimestamp,
```

Comment:
BlockTimestamp shouldn't be used for this. It's somewhat an abuse of the type system. Perhaps use Duration?